packages/clay-core/src/icon-selector/DropdownSelector.tsx

```diff
@@ -0,0 +1,258 @@
+/**
+ * SPDX-FileCopyrightText: © 2025 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import Button from '@clayui/button';
+const spritemap = require('@clayui/css/lib/images/icons/icons.svg');
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import {
+	InternalDispatch,
+	Overlay,
+	useControlledState,
+	useId,
+	useOverlayPosition,
+} from '@clayui/shared';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+
+export type IProps<T> = {
+	/**
+	 * Flag to indicate if menu is showing or not.
+	 */
+	active?: boolean;
+
+	/**
+	 * Custom button component.
+	 */
+	as?:
+		| 'button'
+		| React.ForwardRefExoticComponent<any>
+		| ((props: React.ComponentProps<typeof Button>) => JSX.Element);
+
+	/**
+	 * Value to display the trigger with an icon or text.
+	 */
+	buttonWithIcon?: boolean;
+
+	/**
+	 * The initial value of the active state (uncontrolled).
+	 */
+	defaultActive?: boolean;
+
+	/**
+	 * Property to set the initial value of `items` (uncontrolled).
+	 */
+
+	defaultItems?: Array<T> | null;
+
+	/**
+	 * The initial value of the input (uncontrolled).
+	 */
+	defaultValue?: string;
+
+	/**
+	 * Direction the menu will render relative to the Icon Selector.
+	 */
+	direction?: 'bottom' | 'top';
+
+	/**
+	 * Property to render content with dynamic data.
+	 */
+	items?: Array<T> | null;
+
+	/**
+	 * Funtion to select an icon
+	 */
+	handleIconSelect: (icon: string) => void;
+
+	/**
+	 * Callback for when the active state changes (controlled).
+	 */
+	onActiveChange?: InternalDispatch<boolean>;
+
+	[key: string]: any;
+};
+
+export default function IconSelector<
+	T extends Record<string, any> | string | number
+>({
+	active: externalActive,
+	as: As = Button,
+	buttonWithIcon = false,
+	children,
+	containerElementRef,
+	defaultActive,
+	direction = 'bottom',
+	handleIconSelect,
+	onActiveChange,
+}: IProps<T>) {
+	const menuRef = useRef<HTMLDivElement>(null);
+	const triggerRef = useRef<HTMLButtonElement | null>(null);
+
+	const ariaControls = useId();
```

Comment:
Check that this the more common pattern as opposed to calling this overlayId or something similar.